mono/metadata/marshal.c

```diff
@@ -519,7 +519,11 @@ mono_ftnptr_to_delegate (MonoClass *klass, gpointer ftn)
 
 		if (use_aot_wrappers) {
 			wrapper = mono_marshal_get_native_func_wrapper_aot (klass);
-			this_obj = mono_value_box (mono_domain_get (), mono_defaults.int_class, &ftn);
+			this_obj = mono_value_box_checked (mono_domain_get (), mono_defaults.int_class, &ftn, &error);
+			if (!is_ok (&error)) {
```

Comment:
What if we made mono_error_set_pending_exception return a gboolean?

That way the code can become:
```
if (mono_error_set_pending_exception (&error))
  return NULL
````

Or maybe we could add a return_val_if_nok variant that sets a pending exception.